src/renderer/src/hooks/stores/mapeoDeviceStore.ts

```diff
@@ -0,0 +1,122 @@
+import { create } from 'zustand'
+
+type DeviceType = 'mobile' | 'desktop'
+
+export type Role = 'coordinator' | 'participant' | 'waitingInvite'
+
+type MapeoDevice =
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'waitingInvite'>
+    }
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'coordinator' | 'participant'>
+      dateAdded: Date
+    }
```

Comment:
would break this down since  `waitingInvite` isn't really a role. something along the lines of:

```ts
export type Role = 'coordinator' | 'participant'

type Device = {
  name: string
  deviceId: string
  deviceType: DeviceType
}

type ProjectMember = Device & {
  // Not entirely sure about this field, but thinking it's specific 
  // to only actual project members
  pastDeviceNames: string[] 
  role: Role
  dateAdded: Date
}

// Not sure if an explicit type is needed to reflect an invited device, 
// but in case it's useful...
type InvitedDevice = Device & {
  invited: true
}

type MapeoDevice = ProjectMember | Device | InvitedDevice
```

this is an approximation, so it should be adjusted as needed